__tests__/version.test.ts

```diff
@@ -0,0 +1,239 @@
+      "nightly",
+    ])
+    await expect(resolveVersions(["1.10-nightly"], ".")).resolves.toEqual([
+      "1.10-nightly",
+    ])
+    await expect(resolveVersions(["1.11-nightly"], ".")).resolves.toEqual([
+      "1.11-nightly",
+    ])
+    await expect(resolveVersions(["1.12-nightly"], ".")).resolves.toEqual([
+      "1.12-nightly",
+    ])
+  })
+
+  it("Respects ifMissing error", async () => {
+    await expect(
+      resolveVersions(["1.9-nightly"], ".", { ifMissing: "error" })
+    ).rejects.toThrow("No Julia version exists")
+  })
+
+  it("Respects ifMissing warning", async () => {
+    await resolveVersions(["1.9-nightly"], ".", { ifMissing: "warn" })
+    expect(core.warning).toHaveBeenNthCalledWith(
+      1,
+      expect.stringMatching(/^No Julia version exists/)
+    )
+  })
+
+  it("Default ifMissing behavior", async () => {
+    await expect(resolveVersions(["1.9-nightly"], ".")).rejects.toThrow(
+      "No Julia version exists"
+    )
+  })
+})
+
+describe("resolveVersion tests", () => {
+  // Will need to update these values if `__fixtures__/versions.json` is updated.
+  const latestRelease = "1.11.3"
+  const latestLts = "1.10.8"
+
+  describe("specific versions", () => {
+    it("Must return an available version", () => {
+      expect(resolveVersion("1.0.5", [])).toBeNull()
+      expect(resolveVersion("1.0.5", ["1.0.5", "1.0.6"])).toEqual("1.0.5")
+      expect(resolveVersion("1.0.5", ["1.0.4", "1.0.5"])).toEqual("1.0.5")
+      expect(resolveVersion("1.0.5", ["1.0.4"])).toBeNull()
+      expect(resolveVersion("1.3.0-alpha", [])).toBeNull()
+      expect(
+        resolveVersion("1.3.0-alpha", [
+          "1.2.0",
+          "1.3.0-alpha",
+          "1.3.0-rc1",
+          "1.3.0",
+        ])
+      ).toEqual("1.3.0-alpha")
+    })
+
+    it("Respects v-prefix", () => {
+      expect(resolveVersion("1.0.5", ["v1.0.5", "v1.0.6"])).toEqual("v1.0.5")
+      expect(resolveVersion("1.0.5", ["v1.0.4", "v1.0.5"])).toEqual("v1.0.5")
+      expect(resolveVersion("v1.0.5", ["1.0.5", "1.0.6"])).toEqual("1.0.5")
+      expect(resolveVersion("v1.0.5", ["1.0.4", "1.0.5"])).toEqual("1.0.5")
+    })
+
+    it("version alias: nightly", () => {
+      expect(resolveVersion("nightly", [])).toBeNull()
+    })
+
+    it("version alias: lts", () => {
+      expect(resolveVersion("lts", testVersions)).toEqual(latestLts)
+    })
+  })
+
+  describe("version ranges", () => {
+    it("Chooses the highest available version that matches the input", () => {
+      expect(resolveVersion("1", testVersions)).toEqual(latestRelease)
+      expect(resolveVersion("1.0", testVersions)).toEqual("1.0.5")
+      expect(resolveVersion("^1.3.0-rc1", testVersions)).toEqual(latestRelease)
+      expect(resolveVersion("^1.2.0-rc1", testVersions)).toEqual(latestRelease)
+      expect(resolveVersion("^1.10.0-rc1", testVersions)).toEqual(latestRelease)
+    })
+  })
+
+  // describe("include-prereleases", () => {
```

Comment:
We should un-comment these tests and use something like [test.failing](https://jestjs.io/docs/api#testfailingname-fn-timeout) or [test.skip](https://jestjs.io/docs/api#testskipfailingname-fn-timeout).